serial-proposal.md

```diff
@@ -0,0 +1,71 @@
+Serial proposal
+===
+
+A draft proposal for adding Software and Hardware Serial support to Firmata.
+
+A discussion and sample implementation code for Arduino is available here: https://github.com/firmata/arduino/issues/97
+
+```
+// SW_SERIAL  = 0x00
+// HW_SERIAL1 = 0x01
+// HW_SERIAL2 = 0x02
+// HW_SERIAL3 = 0x03
+
+// config: creates new software Serial or hardware serial instance and calls begin(baud)
+0  START_SYSEX      (0xF0)
+1  SERIAL_DATA      (0x60)  // command byte
+2  CONFIG           (0x11)  // subcommand byte (SERIAL_CONFIG | HW_SERIAL1)
+3  baud             (bits 0 - 6)
+4  baud             (bits 7 - 13)
+5  baud             (bits 14 - 20) // need to send 3 bytes for baud even if value is < 14 bits
+6  bufferLen        (bits 0 - 6)
```

Comment:
Undecided on whether or not to include a buffer length. A variable length buffer would require dynamic memory allocation which may lead to issues on boards with limited memory such as an Arduino Uno. However, a buffer could be useful for any of the newer boards with larger amounts of ram. How to make the buffer optional? A buffer length of zero could indicate that the underlying implementation should use a fixed buffer length (determined by the particular firmware implementation).